From c6b0959255a8c50482f6825b7e27c70496e28359 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 21 Aug 2014 12:58:32 -0400
Subject: [ethernet] qlge: call ql_core_dump() only if dump memory was allocated

Message-id: <1408625927-3701-2-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88524
O-Subject: [PATCH RHEL 7.1 01/16] qlge: call ql_core_dump() only if dump memory was allocated.
Bugzilla: 1089357
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: "malahal@us.ibm.com" <malahal@us.ibm.com>

Bug 1089357 (https://bugzilla.redhat.com/show_bug.cgi?id=1089357)

net commit id 3db9180213af6aa54effc2b94bc9090dc1f68673

Also changed a log message to indicate that memory was not allocated
instead of memory not available!

Signed-off-by: Malahal Naineni <malahal@us.ibm.com>
Acked-by: Jitendra Kalsaria <jitendra.kalsaria@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/qlge/qlge_dbg.c b/drivers/net/ethernet/qlogic/qlge/qlge_dbg.c
index 10093f0..6bc5db7 100644
--- a/drivers/net/ethernet/qlogic/qlge/qlge_dbg.c
+++ b/drivers/net/ethernet/qlogic/qlge/qlge_dbg.c
@@ -740,8 +740,8 @@ int ql_core_dump(struct ql_adapter *qdev, struct ql_mpi_coredump *mpi_coredump)
  int i;
 
  if (!mpi_coredump) {
-  netif_err(qdev, drv, qdev->ndev, "No memory available\n");
-  return -ENOMEM;
+  netif_err(qdev, drv, qdev->ndev, "No memory allocated\n");
+  return -EINVAL;
  }
 
  /* Try to get the spinlock, but dont worry if
diff --git a/drivers/net/ethernet/qlogic/qlge/qlge_mpi.c b/drivers/net/ethernet/qlogic/qlge/qlge_mpi.c
index ff2bf8a..7ad1460 100644
--- a/drivers/net/ethernet/qlogic/qlge/qlge_mpi.c
+++ b/drivers/net/ethernet/qlogic/qlge/qlge_mpi.c
@@ -1274,7 +1274,7 @@ void ql_mpi_reset_work(struct work_struct *work)
   return;
  }
 
- if (!ql_core_dump(qdev, qdev->mpi_coredump)) {
+ if (qdev->mpi_coredump && !ql_core_dump(qdev, qdev->mpi_coredump)) {
   netif_err(qdev, drv, qdev->ndev, "Core is dumped!\n");
   qdev->core_is_dumped = 1;
   queue_delayed_work(qdev->workqueue,
-- 
1.7.1